File: go/tests/test_ipv6_unspec.go

import "math/big"

import "./ipaddress"

    type IPv6UnspecifiedTest struct {
        ip: IPAddress,
        to_s: string,
        to_string: string,
        to_string_uncompressed: string,
        num: big.Int,
    }

    func setup() IPv6UnspecifiedTest {
        return IPv6UnspecifiedTest {
            ip: ipv6_unspec.New(),
            to_s: "::",
            to_string: "::/128",
            to_string_uncompressed: "0000:0000:0000:0000:0000:0000:0000:0000/128",
            num: big.Int.New()
        };
    }

int main() {
  describe("", func() {
    it("test_attributes", func() {
        assert_eq(setup().ip.host_address, setup().num);
        assert_eq(128, setup().ip.prefix().get_prefix());
        assert_eq(true, setup().ip.is_unspecified());
        assert_eq(setup().to_s, setup().ip.to_s());
        assert_eq(setup().to_string, setup().ip.to_string());
        assert_eq(setup().to_string_uncompressed,
                   setup().ip.to_string_uncompressed());
    })
    it("test_method_ipv6", func() {
        assert_eq(true, setup().ip.is_ipv6());
    })
  }
}

File: python/tests/test_ipv6_unspec.py
from ip_address.ipv6_unspec import Ipv6Unspec


class IPv6UnspecifiedTest:
    def __init__(self, ip, to_s, to_string, to_string_uncompressed, num):
        self.ip = ip
        self.to_s = to_s
        self.to_string = to_string
        self.to_string_uncompressed = to_string_uncompressed
        self.num = num


def setup():
    return IPv6UnspecifiedTest(
        ip=Ipv6Unspec.create(),
        to_s="::",
        to_string="::/128",
        to_string_uncompressed="0000:0000:0000:0000:0000:0000:0000:0000/128",
        num=0,
    )


def test_attributes():
    assert setup().ip.host_address == setup().num
    assert 128 == setup().ip.prefix.num
    assert setup().ip.is_unspecified() is True
    assert setup().to_s == setup().ip.to_s()
    assert setup().to_string == setup().ip.to_string()
    assert setup().to_string_uncompressed == setup().ip.to_string_uncompressed()


def test_method_ipv6():
    assert setup().ip.is_ipv6() is True
